### ISSUE TITLE
TRIL tests on z/OS generates non-standard NaN formatting

### ISSUE THREAD

**Issue by sarwat12** [eclipse/omr#7254]
While working on PR #7196 , we noticed that adding **NaN** value handling to the S390 float & double max/min evaluators seems to pass all the [MaxMin.cpp](https://github.com/eclipse/omr/blob/2fb53dccd5a50ede883a06ebf9e3e3ae6cf38c65/fvtest/compilertriltest/MaxMinTest.cpp) tests on Linux on Z, but on z/OS, the tests with NaN operands keeps failing with the following exception:

```
28: [----------] 1444 tests from MaxMin/FloatMaxMin
28: unknown file: Failure
28: Unknown C++ exception thrown in the test body.
28: [  FAILED  ] MaxMin/FloatMaxMin.UsingConst/20, where GetParam() = ((0, NaNQ(1)), ("\x86\x94" "\x81\xA7", 50088187B0)) (0 ms)
28: unknown file: Failure
28: Unknown C++ exception thrown in the test body.
28: [  FAILED  ] MaxMin/FloatMaxMin.UsingConst/21, where GetParam() = ((0, NaNQ(1)), ("\x86\x94\x89\x95", 50088187A0)) (0 ms)
28: unknown file: Failure
28: Unknown C++ exception thrown in the test body.
28: [  FAILED  ] MaxMin/FloatMaxMin.UsingConst/22, where GetParam() = ((0, -NaNQ(1)), ("\x86\x94" "\x81\xA7", 50088187B0)) (1 ms)
28: unknown file: Failure
28: Unknown C++ exception thrown in the test body.
28: [  FAILED  ] MaxMin/FloatMaxMin.UsingConst/23, where GetParam() = ((0, -NaNQ(1)), ("\x86\x94\x89\x95", 50088187A0)) (0 ms)
```

The tests are failing most likely due to non-standard `nan` formatting where the TRIL generator raises an exception:
- `std::snprintf("(dconst %f)", NAN)` yields `(dconst nan)` on Linux/GCC, passing all the tests
- `std::snprintf("(dconst %f)", NAN)` yields `(dconst NaNQ(1))` on zOS/XLC, where the tests fail.

This leads to the Float/Double MaxMin tests currently being disabled on z/OS. Opening up this issue to keep track of potential fixes for the correct `NaN` formatting on z/OS.

**Comment by janvrany** [comment]
This is long known problem. I try to attack it in #5587 "Use custom function to format TRIL code" (not yet merged)

**Comment by janvrany** [comment]
Yes, once my PR #5587 gets in (waiting for review) there would have to be hopefully trivial change needed on z/OS to support NaNs - see https://github.com/eclipse/omr/pull/5587/files#diff-b2afab503b2a84d6f6fa84b75d153dd38e53b5b15b40798548af6f0777a67b5cR467-R469

I decided to do one step at a time, first get make drop-in replacement for std::snprintf() and after that address these details in another PRs.
